### PR TITLE
NewtonsoftJson: only build Newtonsoft.Json.csproj

### PR DESCRIPTION
**Pull request by bobvanderlinden**
###### Motivation for this change

This fixes the build of Newtonsoft.Json, which is blocking other
builds. The tests were not able to compile due to changes in NUnit.
Upgrading Newtonsoft.Json is currently not possible as the project
structure is now using dotnet-core and is incompatible with xbuild.

Example of failing build:

https://hydra.nixos.org/build/81003982

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by samueldr** [comment]
→ #45960 will need this backported (leaving a note for myself).

**Comment by samueldr** [comment]
@GrahamcOfBorg build openra gdata-sharp dotnetPackages.Paket dotnetPackages.FSharpAutoComplete

I have done local builds for `openra`, `gdata-sharp`, `dotnetPackages.Paket` and `dotnetPackages.FSharpAutoComplete` on 18.09. Of those, only `openra` built, but the failures are *unrelated* to this fix. Let's see what the CI sees. (99% sure this gets merged after it finishes building, then backported)

**Comment by bobvanderlinden** [comment]
Awesome, thanks. Note that this just avoids building the unit tests of Newtonsoft.Json. It's not ideal, but I couldn't find another way. I think it won't affect other packages though.

**Comment by samueldr** [comment]
Oh, (*I'm deferring to your knowledge*)

@bobvanderlinden there was an aborted PR earlier, #44478, with commit c6323844e96af9932330f68293c66cc3c2de02e9 upgrading this package. Any comments? If this works, and the test suite passes, and dependencies are happy, it may be better to upgrade?

**Comment by fpletz** [comment]
I can confirm that OpenRA builds and works with https://github.com/NixOS/nixpkgs/commit/c6323844e96af9932330f68293c66cc3c2de02e9. :+1:

**Comment by bobvanderlinden** [comment]
This is related to getting all tests passing for 18.09: #45960

**Comment by samueldr** [comment]
Maybe I wasn't clear enough, let me rephrase:

@bobvanderlinden what's your opinion on using c6323844e96af9932330f68293c66cc3c2de02e9 instead of your PR? Do you see any downsides of upgrading? (I'd guess none, but you seem to be more knowledgeable with .NET stuff.)

**Comment by bobvanderlinden** [comment]
Ah missed your comment. That seems like a better way to go if all packages indeed keep working. :+1:

**Comment by xeji** [comment]
closing in favor of #46679
